pkg/scanner/scanner.go

```diff
@@ -86,6 +108,63 @@ func (s *scannerService) Watch(opts ...ScanRequestOption) {
 	s.requestsQueue.enqueue(req)
 }
 
+func (s *scannerService) WatchDescriptorWallet(
+	desc string,
+	eventType []EventType,
+	blockStart int,
+) error {
+	for _, v := range eventType {
+		switch v {
+		case UnspentUtxo:
+			wallet, err := descriptor.Parse(desc)
+			if err != nil {
+				return err
+			}
+
+			var scripts []descriptor.ScriptResponse
+
+			if wallet.IsRange() {
+				scripts, err = wallet.Script(descriptor.WithRange(numOsScripts))
```

Comment:
not sure to get why a constant number of scripts here ? the idea is to watch a bunch of addresses ? or the "entire" wallet ?

Reply:
The question is what is entire wallet? there must be some final number of addresses we are watching, but i would change here instead of constant to add numOfScripts as an arg to WatchDescriptorWallet, @tiero what do u think?